include/dali/core/mm/detail/free_list.h

```diff
@@ -323,6 +349,185 @@ class coalescing_free_list : public best_fit_free_list {
       *pwhere = blk;
     }
   }
+
+  void merge(coalescing_free_list &&with) {
+    if (!with.head_)
+      return;
+    if (!head_) {
+      swap(with);
+      return;
+    }
+
+    block **a = &head_;
+    block **b = &with.head_;
+    block *new_head = nullptr;
+
+    auto next_block = [&]() {
+      return (*a)->start < (*b)->start ? a : b;
+    };
+
+    block **src = next_block();
+    new_head = *src;
+    *src = (*src)->next;
+    new_head->next = nullptr;
+    block *tail = new_head;
+    while (*a && *b) {
+      src = next_block();
+      block *curr = *src;
+      *src = curr->next;  // advance the source
+      curr->next = nullptr;
```

Comment:
Maybe this could be a function as the same code (almost) is in L369.

Reply:
I thought about it but I could not even come up with a meaningful name for it. To this end, I don't think it would make things even more obfuscated - and I'm quite aware that this code is not easy to read.

Reply:
OK